components/layout/SideNav.tsx

```diff
@@ -460,22 +467,22 @@ const SideNav = ({ sectionLinks, pages, mobileItems }: SideNavProps) => {
           </StickySectionWrap>
 
           <CategoriesWrap>
-            {keys(categories).map((c) => {
+            {sortCategories(activeSection.href, keys(categories)).map((c) => {
               const categoryDocs = categories[c];
               const groups = groupsFrom(categoryDocs);
 
               return (
                 <Category key={c}>
-                  {c && <CategoryHeading>{c}</CategoryHeading>}
+                  {c !== "undefined" && <CategoryHeading>{c}</CategoryHeading>}
 
-                  {groups.map((g) => {
-                    const docs = categoryDocs.filter(byDocGroup(g));
-                    const ungroupedGroup = g === null;
+                  {sortByWeight(groups).map((g?: PageGroup) => {
```

Comment:
Maybe I'm just a little bit blind or not focusing enough, but can you explain how I should add the `weight` to groups to sort them?